agent.go

```diff
@@ -86,6 +90,10 @@ func (c *controller) handleKeyChange(keys []*types.EncryptionKey) error {
 			if key.Subsystem == "networking:gossip" {
 				a.networkDB.SetKey(key.Key)
 			}
+			if key.Subsystem == "networking:gossip" /*"networking:ipsec"*/ {
+				drvEnc.Key = hex.EncodeToString(key.Key)
```

Comment:
I kept the logic we have in discoverapi types of having everything as string or []string.
But yeah, we can probably break that tradition and avoid this unnecessary double conversion.
If you do not mind, I'd address this as part of future bug fixes, taking some time to assess the consequences, it is anyway only affecting internal APIs.